gns3/graphics_view.py

```diff
@@ -1205,6 +1212,31 @@ def captureActionSlot(self):
                 else:
                     QtWidgets.QMessageBox.warning(self, "Capture", "No port available for packet capture on {}".format(node.name()))
 
+    def getCommandLineSlot(self):
+        """
+        Slot to receive events from the get command line action in the
+        contextual menu.
+        """
+
+        items = self.scene().selectedItems()
+        if len(items) != 1:
+            QtWidgets.QMessageBox.critical(self, "Command line", "Please select only one router")
+            return
+        item = items[0]
+        if isinstance(item, NodeItem) and hasattr(item.node(), "commandLine"):
+            router = item.node()
+            if router.commandLine() is None:
+                QtWidgets.QMessageBox.warning(self, "Command line", "Get command line is not supported for this type of node.")
+            elif router.commandLine() == '':
+                QtWidgets.QMessageBox.warning(self, "Command line", "Please start the node in order to get the command line.")
+            else:
+                dialog = QtWidgets.QInputDialog(self)
```

Comment:
The getText static method could be used here. http://doc.qt.io/qt-4.8/qinputdialog.html#getText

We use it in other parts of the code.